network/config.go

```diff
@@ -61,12 +62,28 @@ func DefaultConfig() *Config {
 	}
 }
 
+func validateAddresses(address []string) error {
+	for _, addr := range address {
+		_, err := multiaddr.NewMultiaddr(addr)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // SanityCheck performs basic checks on the configuration.
 func (conf *Config) SanityCheck() error {
 	if conf.EnableRelay {
 		if len(conf.RelayAddrs) == 0 {
 			return errors.Errorf(errors.ErrInvalidConfig, "at least one relay address should be defined")
 		}
 	}
+	if err := validateAddresses(conf.Listens); err != nil {
+		return err
+	}
+	if err := validateAddresses(conf.RelayAddrs); err != nil {
```

Comment:
You can fix the linting issue by: `return validateAddresses(conf.RelayAddrs)`

Reply:
I get an unreachable warning in this case:
```go
// SanityCheck performs basic checks on the configuration.
func (conf *Config) SanityCheck() error {
	if conf.EnableRelay {
		if len(conf.RelayAddrs) == 0 {
			return errors.Errorf(errors.ErrInvalidConfig, "at least one relay address should be defined")
		}
	}
	return validateAddresses(conf.RelayAddrs)
	return validateAddresses(conf.Listens)
}
```

Reply:
Only the last `return validateAddresses(conf.Listens)`
You know it nags that why you created new variable. Just return from it. So it will be like:
```go
if err:= validateAddresses(conf.RelayAddrs); err != nil {
   return err
}
return validateAddresses(conf.Listens) 
```

Reply:
yeah, I got it! that was fixed.